src/main/java/pl/edu/mimuw/chess/ChessBoard.java

```diff
@@ -0,0 +1,179 @@
+
+    for (var piece : blackPlayer.getPieces()) {
+      board[piece.getPosition().y * 2 + 1][piece.getPosition().x * 2 + 1] = piece.getSymbol();
+    }
+    for (var piece : whitePlayer.getPieces()) {
+      board[piece.getPosition().y * 2 + 1][piece.getPosition().x * 2 + 1] = piece.getSymbol();
+    }
+
+    for (int i = 0; i < board.length; i++) {
+      for (int j = 0; j < board[i].length; j++) {
+        if (board[i][j] == '\u0000') {
+          sb.append(CLEAR_BOARD[i][j]);
+        } else {
+          sb.append(board[i][j]);
+        }
+      }
+      sb.append('\n');
+    }
+
+    return sb.toString();
+  }
+
+  /**
+   * Simulates a move for the given player.
+   * @param player the player to simulate a move for.
+   * @return true if the match is over, false otherwise.
+   */
+  private boolean simulateMove(Player player) {
+    final var sb = new StringBuilder();
+
+    if (!player.makeRandomMove(this)) {
+      if (player.getColor() == ChessColor.BLACK) sb.append("Black");
+      else sb.append("White");
+      sb.append(" can't make a move!");
+      System.out.println(sb);
+      return true;
+    }
+
+    System.out.println(this);
+    Util.sleep(SLEEP_LENGTH);
+
+    if (!player.hasAKing()) {
+      if (player.getColor() == ChessColor.BLACK) sb.append("Black");
+      else sb.append("White");
+      sb.append(" lost his king!");
+      System.out.println(sb);
+      return true;
+    }
+
+    Util.clearConsole();
+    return false;
+  }
+
+  /**
+   * @return an array without any pieces on it.
+   */
+  private static char[][] getClearBoardRepresentation() {
+    char[][] result = new char[BOARD_SIZE * 2 + 1][BOARD_SIZE * 2 + 1];
+
+    for (int i = 0; i < result.length; i++) {
+      for (int j = 0; j < result[i].length; j++) {
+        if (i == 0) {
+          if (j == 0) result[i][j] = '╔';
+          else if (j == result[i].length - 1) result[i][j] = '╗';
+          else if (j % 2 == 0) result[i][j] = '╤';
+          else result[i][j] = '═';
+        } else if (i == result.length - 1) {
+          if (j == 0) result[i][j] = '╚';
+          else if (j == result[i].length - 1) result[i][j] = '╝';
+          else if (j % 2 == 0) result[i][j] = '╧';
+          else result[i][j] = '═';
+        } else if (i % 2 == 0) {
+          if (j == 0 || j == result[i].length - 1) result[i][j] = '║';
+          else if (j % 2 == 0) result[i][j] = '┼';
+          else result[i][j] = '─';
+        } else {
+          if (j == 0 || j == result[i].length - 1) result[i][j] = '║';
+          else if (j % 2 == 0) result[i][j] = '│';
+          else if ((i + j) / 2 % 2 == 1) result[i][j] = ' ';
+          else result[i][j] = '█';
+        }
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * @return a new list of black pieces, each of them on their starting position.
+   */
+  private static List<ChessPiece> getNewListOfBlackPieces() {
+    final var result = new LinkedList<ChessPiece>();
```

Comment:
nie ma chyba potrzeby, by był to linked list, starzcyłby pewnie nawet set (tylko wtedy trzeba mieć equlas/hashcode) 😉 btw arraylist standardowo lepiej i nikt by sie nie zastanawiał o co chodziło autorowi (a nie widze naprawde powodu teraz, tym bardziej że z góry znasz rozmiar tego lista)